Variables are not supported in the name field of stages when using Allure Python library

There is an interesting problem when a variable is added to the name field of stages and the Allure Python library is installed.

If you add a variable to the name field of stages, Allure Python doesn't like it at all (see error). It seems like it is trying to resolve it. When Allure is not installed tavern-ci executes the tests just fine. If Allure is installed then you will encounter this issue. The error might be on a different keyword, depending on the variable name that you input. Either you update the title to remove any variables or you will have to remove the Allure Python module. 

Example test_product.tavern.yml:

```
test_name: Add products for tests

stages:
  - name: Add product one {service.host}
    request:
      url: '{service.url}/product/'
```

Error output by taver-ci where Allure is complaining about not finding the variable provided in the stage name (i.e., service):
```python
...
                partial = functools.partial(
                    run_stage_with_retries, sessions, stage, test_block_config
                )
    
                allure_name = "Stage {}: {}".format(idx, stage["name"])
                step = wrap_step(allure_name, partial)
    
                try:
>                   step()

/pyenv-allure/lib/python3.9/site-packages/tavern/core.py:189: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

a = (), kw = {}, __tracebackhide__ = True, params = OrderedDict(), args = []

    @wraps(func)
    def impl(*a, **kw):
        __tracebackhide__ = True
        params = func_parameters(func, *a, **kw)
        args = list(map(lambda x: represent(x), a))
>       with StepContext(self.title.format(*args, **params), params):
E       KeyError: 'service'

/pyenv-allure/lib/python3.9/site-packages/allure_commons/_allure.py:171: KeyError
...
```

--- a/tavern/core.py
+++ b/tavern/core.py
@@ -10,6 +10,7 @@ import pytest
 
 from tavern.schemas.files import wrapfile
 from tavern.util.strict_util import StrictLevel
+
 from .plugins import get_expected, get_extra_sessions, get_request_type, get_verifiers
 from .testutils.pytesthook import call_hook
 from .util import exceptions

--- a/tavern/core.py
+++ b/tavern/core.py
@@ -10,7 +10,6 @@ import pytest
 
 from tavern.schemas.files import wrapfile
 from tavern.util.strict_util import StrictLevel
-
 from .plugins import get_expected, get_extra_sessions, get_request_type, get_verifiers
 from .testutils.pytesthook import call_hook
 from .util import exceptions
@@ -182,7 +181,9 @@ def run_test(in_file, test_spec, global_cfg):
                 run_stage_with_retries, sessions, stage, test_block_config
             )
 
-            allure_name = "Stage {}: {}".format(idx, stage["name"])
+            allure_name = "Stage {}: {}".format(
+                idx, format_keys(stage["name"], test_block_config["variables"])
+            )
             step = wrap_step(allure_name, partial)
 
             try:

--- a/tests/integration/test_allure.tavern.yaml
+++ b/tests/integration/test_allure.tavern.yaml
@@ -1,0 +1,16 @@
+---
+
+test_name: "Test test name can have formatting in it: {host}"
+
+includes:
+- !include common.yaml
+
+stages:
+- name: "Test stage name can have formatting in it: {host}"
+  request:
+    url: "{host}/echo"
+    method: POST
+    json: hi
+  response:
+    status_code: 200
+    json: hi

--- a/tox-integration.ini
+++ b/tox-integration.ini
@@ -19,6 +19,7 @@ changedir =
 deps =
     docker-compose
     flask
+    allure-pytest
     pyjwt
     pytest-xdist
     pytest-cov

